core/include/drivers/imsic.h

```diff
@@ -0,0 +1,72 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (c) 2021 Western Digital Corporation or its affiliates.
+ * Copyright (c) 2022 Ventana Micro Systems Inc.
+ * Copyright (c) 2025 Beijing Institute of Open Source Chip (BOSC)
+ *
+ * Authors:
+ *   Anup Patel <anup.patel@wdc.com>
+ *   Huang Borong <huangborong@bosc.ac.cn>
+ */
+
+#ifndef __DRIVERS_IMSIC_H
+#define __DRIVERS_IMSIC_H
+
+#include <kernel/interrupt.h>
+#include <tee_api_defines.h>
+#include <types_ext.h>
+
+struct imsic_data {
```

Comment:
For future readers, could you add an inline description comment like
```c
/*
 * struct imsic_data - IMISC interrupt controller
 * @imisc_base: IOmem base address
 * @size: IOmem address range size
 * ...
```
It would be nice `struct aplic_data` gets also described.
Can be addressed later. Not blocking for the series.